app_dart/lib/src/request_handlers/push_build_status_to_github.dart

```diff
@@ -28,79 +30,95 @@ class PushBuildStatusToGithub extends ApiRequestHandler<Body> {
     @visibleForTesting DatastoreServiceProvider datastoreProvider,
     @visibleForTesting LoggingProvider loggingProvider,
     @visibleForTesting BuildStatusProvider buildStatusProvider,
+    @visibleForTesting
+        this.branchHttpClientProvider = Providers.freshHttpClient,
+    @visibleForTesting this.gitHubBackoffCalculator = twoSecondLinearBackoff,
   })  : datastoreProvider =
             datastoreProvider ?? DatastoreService.defaultProvider,
         loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
         buildStatusProvider =
             buildStatusProvider ?? const BuildStatusProvider(),
+        assert(branchHttpClientProvider != null),
```

Comment:
Do you have any reference to this point? I see assert has been used widely in cocoon backend.

Reply:
Since we are having default values for parameters, `@required` doesn't seem to be suitable.
```The type parameter 'branchHttpClientProvider' is annotated with @required but only named parameters without a default value can be annotated with it.```

Reply:
Error on my part then. I didn't see the default assignment section.